packages/babel-core/src/config/config-chain.js

```diff
@@ -148,6 +172,43 @@ export function buildRootChain(
   };
 }
 
+function babelrcLoadEnabled(
+  context: ConfigContext,
+  babelrc: BabelrcSearch,
+): string | void {
+  if (babelrc === false) return undefined;
+
+  const filename = context.filename;
+  if (filename === null) {
+    throw new Error(
+      `Attempted to search for .babelrc, but no filename was given. ` +
+        `To disable .babelrc searching, pass babelrc: false to Babel.`,
+    );
+  }
+
+  const roots = Array.isArray(babelrc) ? babelrc : [babelrc];
+
+  const searchEnabled = roots.some(root => {
+    const pattern = path.resolve(context.cwd, root);
+
+    const match = micromatch.capture(path.resolve(pattern, "**"), filename);
+
+    if (match) {
+      const inDependency =
+        match
+          .pop()
+          .split(path.sep)
+          .indexOf("node_modules") !== -1;
+
+      if (!inDependency) return true;
+    }
+
+    return !!micromatch.capture(path.pattern, filename);
```

Comment:
What is `path.pattern`?

Reply:
Looks like just a bug. Should be `pattern`.